Makefile

```diff
@@ -1,3 +1,15 @@
+# If JOBS environment variable not defined, default to the number of CPUs available
+ifeq ($(JOBS),)
+JOBS:=1
+OS:=$(shell uname -s)
+ifeq ($(OS),Linux)
+  JOBS:=$(shell grep -c ^processor /proc/cpuinfo)
```

Comment:
If you want somewhere to copy&paste from, try [ProcessorCount.cmake](https://github.com/Kitware/CMake/blob/master/Modules/ProcessorCount.cmake), where we made that work for all platforms we could get access to.